src/plugins/implementations/FilterPlugin.tsx

```diff
@@ -0,0 +1,177 @@
+    if(!this.toParameterName){
+      throw new Error("ToParameterName attribute of the FilterPlugin is not set")
+    }
+    if(!this.fromParameterName){
+      throw new Error("FromParameterName attribute of the FilterPlugin is not set")
+    }
+    this.setParameters();
+  }
+
+  private setParameters() {
+    const parameters: { [key: string]: string } = {};
+    parameters[this.fromParameterName!] = toOrigamServerString(this.dateFrom);
+    parameters[this.toParameterName!] = toOrigamServerString(this.dateTo);
+    this.setFormParameters?.(parameters);
+  }
+
+  getComponent(data: IPluginData): JSX.Element {
+    let localizer = new Localizer(localizations, "en-US");
+    for (let timeUnit of this.timeUnits) {
+      timeUnit.label = localizer.translate(timeUnit.value)
+    }
+    return <FilterComponent
+      filterPlugin={this}
+      localizer={localizer}/>;
+  }
+
+  addTime(args:{start: Moment}){
+    switch(this.selectedTimeUnit.value){
+      case "month":
+        this.dateFrom = moment([ args.start.year(), args.start.month()])
+        this.dateTo =  this.dateFrom.clone().add(1, 'months');
+        break;
+      case "day":
+        this.dateFrom = moment([ args.start.year(), args.start.month(), args.start.date()])
+        this.dateTo = this.dateFrom.clone().add(1, 'days');
+        break;
+      case "hour":
+        this.dateFrom = moment([ args.start.year(), args.start.month(), args.start.date(), args.start.hour()])
+        this.dateTo = this.dateFrom.clone().add(1, 'hours');
+        break;
+      default:
+        throw new Error("time unit \"" + this.selectedTimeUnit.value + "\" not implemented" )
+    }
+  }
+
+  subtractTime(args:{end: Moment}){
+    switch(this.selectedTimeUnit.value){
+      case "month":
+        this.dateTo = moment([ args.end.year(), args.end.month() ])
+        this.dateFrom = this.dateTo.clone().add(-1, 'months')
+        break;
+      case "day":
+        this.dateTo = moment([ args.end.year(), args.end.month(), args.end.date() ])
+        this.dateFrom = this.dateTo.clone().add(-1, 'days')
+        break;
+      case "hour":
+        this.dateTo = moment([ args.end.year(), args.end.month(), args.end.date(), args.end.hour()])
+        this.dateFrom = this.dateTo.clone().add(-1, 'hours')
+        break;
+      default:
+        throw new Error("time unit \"" + this.selectedTimeUnit.value + "\" not implemented" )
+    }
+  }
+
+  async nextIntervalClick() {
+    this.addTime({start: this.dateTo});
+    await this.refresh();
+  }
+
+  async previousIntervalClick() {
+    this.subtractTime({end: this.dateFrom});
+    await this.refresh();
+  }
+
+  async setTimeunit(timeUnit: IOption<string>){
+    this.selectedTimeUnit = timeUnit;
+    this.addTime({start: this.dateFrom});
+    await this.refresh();
+  }
+}
+
+@observer
+class FilterComponent extends React.Component<{
+  filterPlugin: FilterPlugin,
+  localizer: Localizer
+}> {
+
+  plugin = this.props.filterPlugin;
```

Comment:
This will be set once before the constructor gets called. I expect `filterPlugin` is already known and constant, but generally better would be:

```suggestion
  get plugin() {
      return this.props.filterPlugin;
  }
```

Reply:
This was ment to be just a shorthand. Why is it better to use a getter here? Do you mean that some could try to change it while it should be a constant?
whar about:
 private readonly plugin = this.props.filterPlugin;